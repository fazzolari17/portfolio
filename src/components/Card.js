const Card = ({children, style, invert, className }) => {

  const background = invert ? 'var(--header-contrastColor, #75ABBC)' : 'var(--hero-background, #395E66)';
  const textColor = invert ? 'var(--header-background, #071013)' : 'var(--header-text, #F0F8FF)';
  
  const componentStyle = {
    fontFamily: 'Lexend Deca, sans-serif',
    padding: '2rem',
    margin: '2rem auto',
    maxWidth: 'fit-content',
    backgroundColor: background,
    color: textColor,
    borderRadius: '5px',
    boxShadow: '-1px 0px 38px 8px rgba(0,0,0,0.51)',
  }
  
  return (
    <section style={{ ...componentStyle, ...style }} className={className}>
      {children}
    </section>
  )
};

export default Card;