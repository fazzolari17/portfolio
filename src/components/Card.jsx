import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ children, style, invert, className, shadow }) => {

  const background = invert ? 'var(--header-contrastColor, #75ABBC)' : 'var(--hero-background, #395E66)';
  const textColor = invert ? 'var(--header-background, #071013)' : 'var(--header-text, #F0F8FF)';
  const boxShadow = shadow ? '-1px 0px 38px 8px rgba(0,0,0,0.51)' : '';

  const componentStyle = {
    fontFamily: 'Lexend Deca, sans-serif',
    padding: '2rem',
    margin: '2rem auto',
    maxWidth: 'fit-content',
    backgroundColor: background,
    color: textColor,
    borderRadius: '5px',
    boxShadow: boxShadow,
  };

  return (
    <section style={{ ...componentStyle, ...style }} className={className}>
      {children}
    </section>
  );
};

Card.propTypes = {
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  children: PropTypes.node,
  invert: PropTypes.bool,
  className: PropTypes.string,
  shadow: PropTypes.bool
};

export default Card;