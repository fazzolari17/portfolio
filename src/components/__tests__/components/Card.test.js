import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../Card';

describe('Card Component', () => {
  describe('Snapshot', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <Card />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('card component unit tests', () => {
    beforeEach(() => {
      render(
        <Card className={'test-class'} style={{ background: 'red' }}>
          <p>Child Component</p>
        </Card>
      );
    });
    it('renders children', () => {
      const childComponent = screen.getByText(/child component/i);
      expect(childComponent).toBeInTheDocument();
    });

    it('passes the className through to component', () => {
      const childComponent = screen.getByText(/child component/i);
      expect(childComponent.parentElement).toHaveClass('test-class');
    });

    it('passes the style attributes through to component', () => {
      const childComponent = screen.getByText(/child component/i);
      expect(childComponent.parentElement).toHaveStyle('background: red');
    });

  });
});