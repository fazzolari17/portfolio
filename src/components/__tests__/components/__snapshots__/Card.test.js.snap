// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Snapshot renders correctly 1`] = `
<section
  style={
    Object {
      "backgroundColor": "var(--hero-background, #395E66)",
      "borderRadius": "5px",
      "boxShadow": "",
      "color": "var(--header-text, #F0F8FF)",
      "fontFamily": "Lexend Deca, sans-serif",
      "margin": "2rem auto",
      "maxWidth": "fit-content",
      "padding": "2rem",
    }
  }
/>
`;
